apps/desktop/src/components/v3/GiphyPlugin.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+	import { getCursorPosition, insertImageAtCaret } from '$lib/textEditor/selection';
+	import { debounce } from '$lib/utils/debounce';
+	import { gifPaginator, GiphyFetch } from '@giphy/js-fetch-api';
+	import { Gif } from '@giphy/svelte-components';
+	import Icon from '@gitbutler/ui/Icon.svelte';
+	import { clickOutside } from '@gitbutler/ui/utils/clickOutside';
+	import { portal } from '@gitbutler/ui/utils/portal';
+	import {} from '@lexical/markdown';
+	import { onMount } from 'svelte';
+	import { getEditor } from 'svelte-lexical';
+	import type { IGif } from '@giphy/js-types';
+
+	const editor = getEditor();
+
+	let position: { left: number; top: number } | undefined = $state();
+
+	onMount(() => {
+		const root = editor.getRootElement();
+		root?.addEventListener('keydown', monitorKeystrokes);
+		return () => {
+			root?.removeEventListener('keydown', monitorKeystrokes);
+		};
+	});
+
+	let query = $state('');
+	let loading = $state(false);
+	let loadingDiv: HTMLDivElement | undefined = $state();
+
+	/** Key sequence that will show the giphy search box. */
+	const matchSequence = '/gif';
+
+	let inputBuffer = '';
+	let prefixMatched = false;
+	let queryBuffer = '';
+
+	function monitorKeystrokes(e: KeyboardEvent) {
+		const key = e.key;
+		if (!prefixMatched) {
+			if (key.length === 1 && key.match(/[\w /]/)) {
+				inputBuffer += key;
+				if (inputBuffer.endsWith('/gif ')) {
```

Comment:
What happens if I'm typing /gif somewhere in the middle of my commit message?

Reply:
Nothing will happen unless you hit enter after first typing a valid string.